src/server/game/Entities/Creature/Creature.cpp

```diff
@@ -3608,6 +3619,22 @@ void Creature::InitializeInteractSpellId()
         SetInteractSpellId(0);
 }
 
+void Creature::InitializeSummonInfo(SummonInfoArgs const& args)
+{
+    ASSERT(_summonInfo == nullptr, "SummonInfo has already been initialized and must not be allocated again!");
+    _summonInfo = std::make_unique<SummonInfo>(this, args);
+}
+
+SummonInfo* Creature::GetSummonInfo() const
+{
+    return _summonInfo.get();
+}
+
+bool Creature::IsSummon() const
```

Comment:
To improve API usability, we could use GetSummonInfo() directly to return nullptr when the creature is not a summon, rather than having IsSummon and GetSummonInfo.

Reply:
This is a convenience helper to improve readability. Also GetSummonInfo already does return nullptr when not set because std::unique_ptr::get() returns the plain raw pointer to the object.